### PR TITLE
Add global unique identifiers

### LINKED ISSUE #606: Simplify LTI/CWL login
- Allow customization of unique identifier for CAS/SAML
- Allow shared unique identifier between lti and CAS/SAML
  - simplify signup process for lti users
  - allow automatic enrollment into lti courses of CAS/SAML users who have never followed lti link

### PR DESCRIPTION
**Pull request by andrew-gardener**
Simplifies LTI and CAS/SAML logins by automatically creating accounts on login (when global unique identifier is set for that login method). When account is created, global unique identifier will be associated with the user so that when logging in with a different method, account can be automatically linked.

If the LTI consumer does not use global unique identifiers, then it can be linked to a compair account though login (via username/password or CAS/SAML) as per normal.

- Simplified how xapi external actor accounts are handled. Now when properly setup will use global unique identifier (no need to check `third_party_user` or `lti_user` tables)
- Changed `lti_consumer`'s`user_id_override` to `global_unique_identifier_param` and removed all `lti_user` records linked to consumer that used it. Users will automatically be re-linked to these accounts on their next lti launch or membership refresh since the old lti user_id field is automatically stored as the user's global unique identifier. this change was done so that user id would be properly stored for using the LTI Assignment and Grades Service later on.
- Added user management function `generate_global_unique_identifiers` to automatically populate global unique identifiers based on existing CAS, SAML, and/or LTI links (requires having environment variables already setup for CAS and SAML and should not be needed for LTI due to migration but might be helpful anyways).
- Added Index on `comparable` in `answer` table
- Also fix acceptance tests

Closes #606

### CODE REVIEW

File: compair/api/login.py [hunk 1]
```diff
@@ -156,17 +156,17 @@ def cas_auth():
...
 
             authenticate(thirdpartyuser.user, login_method=thirdpartyuser.third_party_type.value)
             thirdpartyuser.params = validation_response.attributes
 
             if sess.get('LTI') and sess.get('lti_create_user_link'):
                 lti_user = LTIUser.query.get_or_404(sess['lti_user'])
-                lti_user.compair_user = thirdpartyuser.user
+                lti_user.compair_user_id = thirdpartyuser.user_id
                 lti_user.upgrade_system_role()
                 lti_user.update_user_profile() # only one update_user_profile needed TODO: simplify this in #606
```
Comment (kitsook) [inline compair/api/login.py]: Is the `TODO` still needed before closing #606?

File: alembic/versions/04529827d9af_add_global_unique_identifier.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    # rename user_id_override to global_unique_identifier_param
+    with op.batch_alter_table('lti_consumer', naming_convention=convention) as batch_op:
+        batch_op.alter_column('user_id_override', new_column_name='global_unique_identifier_param',
+            nullable=True, existing_type=sa.String(255))
+
+    with op.batch_alter_table('lti_user', naming_convention=convention) as batch_op:
+        batch_op.add_column(sa.Column('global_unique_identifier', sa.String(length=255), nullable=True))
+
+    with op.batch_alter_table('user', naming_convention=convention) as batch_op:
+        batch_op.add_column(sa.Column('global_unique_identifier', sa.String(length=255), nullable=True))
```
[the remark this answers is not in the document]
Comment (andrew-gardener) [reply alembic/versions/04529827d9af_add_global_unique_identifier.py]: It technically doesn't need to be in `lti_user` but I've been trying to store all relevant launch information in the lti tables to help debug issues if things get out of sync or weird for whatever reason.

The strategy I used for handling global unique identifiers in both LTI and CAS/SAML is to try and not touch existing user account links except though the user manage function. I can see that maybe setting it for existing users who have blank `global_unique_identifier` in the user table on login might be a good idea (need to mull it over some more)

File: compair/api/login.py (hunk 1 above)
Comment (andrew-gardener) [reply compair/api/login.py]: no, I think I covered what I wanted simplified in this PR and #760

Comment (andrew-gardener) [reply compair/api/login.py]: Actually I forgot to add an `upgrade_system_role` to for CAS/SAML login so I'll be adding that as well

**Pull request merged**